### PR TITLE
Update SerializationService to bring it in line with Java client

### PR DESCRIPTION
**Pull request by cangencer**
Update default serializers to be constant, and add serializers for List
and LinkedList

### CODE REVIEW

File: Hazelcast.Net/Hazelcast.IO.Serialization/StreamSerializerAdapter.cs [hunk 1]
```diff
@@ -54,29 +52,6 @@ public ISerializer GetImpl()
             return _serializer;
         }
 
-        public override bool Equals(object o)
```
Comment (asimarslan) [inline Hazelcast.Net/Hazelcast.IO.Serialization/StreamSerializerAdapter.cs]: Don't we need Equal and HashCode methods ?

**Comment by asimarslan** [comment]
What about putting the .Net default serialization in the last step as Java did. It won't be compatible with other clients but would work perfectly for .Net clients.

File: Hazelcast.Net/Hazelcast.IO.Serialization/StreamSerializerAdapter.cs (hunk 1 above)
Comment (cangencer) [inline Hazelcast.Net/Hazelcast.IO.Serialization/StreamSerializerAdapter.cs]: I removed them becausd they were not in any code path (never used)

**Comment by cangencer** [comment]
I've added support for C# Serializable now, which uses type ID -101, which probably should be added to the Java list of constants so that it's reserved.

**Comment by hasancelik** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by asimarslan** [comment]
:+1:

**Pull request merged**